Homework_Concurrency/src/main/java/FestivalStatisticThread.java

```diff
@@ -0,0 +1,28 @@
+import java.util.concurrent.TimeUnit;
+
+public class FestivalStatisticThread extends Thread{
+
+    Gate gate;
+
+    public FestivalStatisticThread(Gate gate) {
+        this.gate = gate;
+        GetStats();
+    }
+
+    public synchronized void GetStats(){
```

Comment:
the synchronization is in the Gate, not on the Stats (since here we have one thread), when adding and checking size we must have a sync queue